File: app/src/test/java/io/github/tobyhs/weatherweight/AppTest.java
package io.github.tobyhs.weatherweight;

import androidx.test.core.app.ApplicationProvider;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.CoreMatchers.notNullValue;
import static org.hamcrest.MatcherAssert.assertThat;

@RunWith(RobolectricTestRunner.class)
@Config(application = App.class)
public class AppTest {
    private App app;

    @Before
    public void setup() {
        app = ApplicationProvider.getApplicationContext();
    }

    @Test
    public void applicationInjector() {
        assertThat(app.applicationInjector(), is(notNullValue()));
    }
}

File: app/src/test/java/io/github/tobyhs/weatherweight/AppTest.kt
package io.github.tobyhs.weatherweight

import androidx.test.core.app.ApplicationProvider

import org.hamcrest.CoreMatchers.instanceOf
import org.hamcrest.MatcherAssert.assertThat

import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

@RunWith(RobolectricTestRunner::class)
@Config(application = App::class)
class AppTest {
    private lateinit var app: App

    @Before
    fun setup() {
        app = ApplicationProvider.getApplicationContext()
    }

    @Test
    fun applicationInjector() {
        assertThat(app.applicationInjector(), instanceOf(DaggerAppComponent::class.java))
    }
}
